dev-docs/RFCs/v7.0/google-maps-integration.md

```diff
@@ -0,0 +1,120 @@
+
+deck.gl was initially designed to work with mapbox-gl. Its MapView is implemented to precisely match the camera behavior of Mapbox. So far, Mapbox has been the only base map solution for deck.gl. There has been multiple attempts from external developers to use deck.gl with Google Maps API. It is no trivial task, because Google Maps is much more opaque comparing to mapbox-gl.
+
+This RFC proposes a new module `@deck.gl/google-maps` that handles the synchronization between Deck and Google Maps. This offering will make it a lot easier for Google Maps devlopers to leverage deck.gl for data visualization.
+
+## Proposal
+
+### DeckOverlay
+
+The proposed `@deck.gl/google-maps` module will have a single export `DeckOverlay`. This class implements the [OverlayView](https://developers.google.com/maps/documentation/javascript/reference/#OverlayView) interface and can be used as any other Google Maps overlay:
+
+```js
+  import DeckOverlay from '@deck.gl/google-maps';
+  import {GeoJsonLayer} from '@deck.gl/layers';
+
+  // Create map
+  const map = new google.maps.Map(document.getElementById('map'), {
+    mapTypeId: 'hybrid',
+    center: { lat: 40, lng: -100 },
+    zoom: 5,
+    tilt: 0
+  });
+
+  // Create overlay instance
+  const overlay = new DeckOverlay({
+    layers: [
+      new GeoJsonLayer({
+        ...
+      })
+    ]
+  });
+  // Add overlay to map
+  overlay.setMap(map);
+```
+
+##### `Constructor`
+
+`const deckOverlay = new DeckOverlay(props)`
+
+Parameters:
+
+* `props.api` - the client from which we pull the Maps API classes. Default `google.maps`.
+* The following [Deck](/docs/api-reference/deck.md) props are supported:
+  - `layers`
+  - `layerFilter`
+  - `effects`
+  - `parameters`
+  - `pickingRadius`
+  - `useDevicePixels`
+  - `onWebGLInitialized`
+  - `onBeforeRender`
+  - `onAfterRender`
+  - `onLoad`
+
+
+##### `setMap`
+
+`deckOverlay.setMap(map)`
+
+Add/remove the overlay from a map.
+
+##### `setProps`
+
+`deckOverlay.setProps(props)`
+
+Update (partial) props.
+
+##### `pickObject`
+
+Equivalent of `deck.pickObject`.
+
+##### `pickObjects`
+
+Equivalent of `deck.pickObjects`.
+
+##### `pickMultipleObjects`
+
+Equivalent of `deck.pickMultipleObjects`.
+
+##### `finalize`
+
+Remove the overlay and release all resources.
+
+
+## Cost, Risks, Open Questions
+
+### How aligned should the API be with the mapbox module?
+
+While `MapboxLayer` from `@deck.gl/mapbox` is a wrapper of a single layer, the `DeckOverlay` class is a wrapper of `Deck`. The `DeckOverlay` API is designed to align with 1) the Google Maps overlay API, and 2) the Deck class. 
+
+This module will be much simpler than `@deck.gl/mapbox`. Because there is no context sharing, there is no need for WebGL state management, custom rendering order, etc. All layers are rendered once the overlay is redrawn.
+
+### What Deck features are supported?
+
+Supported features:
+
+- Layers
+- Effects
+- Auto-highliting
+- Transitions / Animations
+- Interaction callbacks
```

Comment:
Unless there's another way to keep the DeckGL canvas synced to the Maps API DOM than the one I've used in [this prototype](https://gist.github.com/donmccurdy/d3aaedf98bd05a1b4a6bbd563804611f#file-googlemapslayer-ts-L95-L99), the canvas will not directly receive any DOM events unless we specifically re-emit them based on Maps API events.

Which interaction callbacks do you mean here? Certainly things like `onLayerHover` and `onLayerClick` seem helpful. But I don't think the layer needs to replicate `onDrag[Start|End]`, which would already be provided by Maps API.

Should the interaction callbacks use the Maps API `addListener` pattern, or something more `EventTarget`-like?

Reply:
Good call, drag events are not going to be supported.